squad_client/shortcuts.py

```diff
@@ -61,21 +61,25 @@ def submit_results(group_project_slug=None, build_version=None, env_slug=None, t
     testrun.attachments = attachments
     testrun.environment = environment
 
-    for test_name in tests.keys():
-        test = Test()
-        test.name = test_name
-        testrun.add_test(test)
-
-        if type(tests[test_name]) is dict:
-            test.status = tests[test_name].get('result')  # raise error if result is invalid
-            test.log = tests[test_name].get('log')
-        else:
-            test.status = tests[test_name]
-
-    for metric_name in metrics.keys():
-        metric = Metric()
-        metric.name = metric_name
-        metric.result = metrics[metric_name]
-        testrun.add_metric(metric)
-
-    testrun.submit_results()
+    if tests:
```

Comment:
Is this necessary? tests.keys() will produce empty 'list' (it's not a real list in p3) so the loop will finish before 1st iteration. IMHO this check is redundant.

Reply:
I remembered why I added this. In "commands.submit()", `results_dict` is initialized to `None` and if no tests are given, it'll remain the same and passed to "shortcuts.submit_results()", making tests=None thus braking when calling tests.keys(). Perhaps we can just initialize `results_dict` to an empty dict?

Reply:
yes, we can initialize results_dict to empty dict.

Reply:
done